Manual migration

skyfox93 opened the pull request:
<!--
*Before merging your pull request, be sure that the follow steps have been followed:*

1. Your pull request is up-to-date with master
2. You've selected "Squash and merge" option as the merge method (default)
3. Review merge commit message and remove unnecessary lines
4. make sure you have linked the issue that this PR addresses, see right sidebar ->
-->

## Dev Summary
<!-- Link to the mocks, with node-id if possible! Note you must be signed in for Figma to update the URL with the node-id
  i.e. Messaging Inbox node: https://www.figma.com/file/fYkreARKdHsMh8iPSh7dti/GC-prototype-v1?node-id=0%3A453
  Full mocks: https://www.figma.com/file/fYkreARKdHsMh8iPSh7dti/GC-prototype-v1
-->
### To run a migration manually -
import it and list it in the migrations array in the DataSourceConnectionService
call `npm run typeorm:migration-run` to run migrations

### required for transition
Because the existing schema is not reflected in the migrations table, the following ONE TIME step is nessesary to migrate the database to using manual migrations:
call `npm run typeorm:schema-drop` to clear the current database
call `npm run typeorm:migration-run` to make the database and migration table reflect the existing migrations

## Additional requirements
### required everytime your database is out of date with main
 `npm run typeorm:migration-run` to bring it up to date

### required every time you make changes to an entity on a feature branch
make sure you are migrated up to the existing main:  call `npm run typeorm:migration-run`
call `npm run typeorm:migration-generate` to generate a migration based on the changes you've made
call `npm run typeorm:migration-run` to run the migration

### required everytime before moving back to main after running a migration on a feature branch
`npm run typeorm:migration-revert` to migrate back "down" to main ( if you've added multiple migrations, you must revert multiple times)

### required when merging a pull request into that has migrations
- make sure there were no conflicting or duplicate migrations entered since the pull request was made
- if there were, or if you wish to be safe before merge:

    - on the main branch, make sure your database is up to date and you've , run all pending migrations, then run `npm run typeorom:migration-generate`. It should fail, as you have no made any changes to main.  If migrations are generated when no changes have been made to main, it means that either a) theres a problem with main, or main is missing a migration or b) you made migrations on another branch and forgot to revert them

   - checkout the feature branch, discard any migration files in the pull request and run `npm run typeorm:migration-generate` to re-generate them against the existing main
   - `npm run typeorm:migration-run` and test the pull request works
   - `npm run typeorm:migration-revert` before moving back into main




the following package scripts have been added: 
```
 "typeorm:migration-run": "npm run typeorm migration:run -- -d ./src/datasource.ts",
        "typeorm:migration-generate": gener./migrations/$npm_config_name",
        "typeorm:migration-create": "npm run typeorm -- migration:create ./migrations/$npm_config_name",
        "typeorm:migration-revert": "npm run typeorm -- -d ./src/datasource.ts migration:revert",
        "typeorm:schema-drop": "npm run typeorm -- -d ./src/datasource.ts schema:drop",
        "typeorm:schema-sync": "npm run typeorm -- -d ./src/datasource.ts schema:sync",
```
<!--
  A detailed outline of what this commit includes, such as:
  benefits to user/product side
  why you architected your code this way, what other options there were and why this is best
-->


## Test Plan
<!--
A test plan that you followed to confirm this commit works as intended, and
that it does not break any existing changes.

You should include screenshots to help the reviewer understand what to look for when visual
changes are includes. GIFs help immensely when explaining interactions and animations.
-->

esteban-gs reviewed (changes requested):
@skyfox93 thanks for putting this together. It's going to work really well with our seed script. One small change would be to update the `README.md` with your notes in the PR description and another small change in the comments.

esteban-gs commented on server/package.json:
```diff
@@ -22,6 +22,13 @@
+        "typeorm:migration-revert": "npm run typeorm -- -d ./src/datasource.ts migration:revert",
+        "typeorm:schema-drop": "npm run typeorm -- -d ./src/datasource.ts schema:drop",
+        "typeorm:schema-sync": "npm run typeorm -- -d ./src/datasource.ts schema:sync",
         "seed:drop": "ts-node -r tsconfig-paths/register src/database/seeding/seed.ts"
```
While we're looking at scripts, could you rename the `seed:drop` command to another name similar to what you named the other commands and also update README.md

skyfox93 commented:
Updated the readme with a migrations section and renamed `seed:drop` to `db:seed`. (Seeding does not use the typeorm cli so `typeorm:seed` is inappropriate)

esteban-gs commented:
@skyfox93 , we should be good to merge once we update with latest main version.
